src/platform/Zephyr/BLEAdvertisingArbiter.h

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/Span.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/sys/slist.h>
+
+#include <cstdint>
+
+namespace chip {
+namespace DeviceLayer {
+
+/**
+ * @brief Bluetooth LE advertising arbiter
+ *
+ * The purpose for this class is to coordinate BLE advertising between different application
```

Comment:
Maybe we should mention that the API enforces the user to guarantee that the life time of passed request object is sustained throughout the usage of the `BLEAdvertisingArbiter`.

Reply:
It is already documented a few lines below (before `InsertRequest` method).

Reply:
I know, this is just a nit, but is applies also to the `CancelRequest`, right?. We may just explicitly inform the user that caller is the owner of the passed request objects. But it's it's up to you.